src/core/Song.cpp

```diff
@@ -668,20 +644,18 @@ void Song::stop()
 		case Timeline::StopBehaviour::BackToZero:
 			if (m_playMode == PlayMode::MidiClip)
 			{
-				getPlayPos().setTicks(std::max(0, -m_midiClipToPlay->startTimeOffset()));
+				timeline.setTicks(std::max(0, -m_midiClipToPlay->startTimeOffset()));
 			}
 			else
 			{
-				getPlayPos().setTicks(0);
+				timeline.setTicks(0);
 			}
-			m_elapsedMilliSeconds[static_cast<std::size_t>(m_playMode)] = 0;
 			break;
 
 		case Timeline::StopBehaviour::BackToStart:
 			if (timeline.playStartPosition() >= 0)
 			{
-				getPlayPos().setTicks(timeline.playStartPosition().getTicks());
-				setToTime(timeline.playStartPosition());
+				timeline.setTicks(timeline.playStartPosition().getTicks());
 
 				timeline.setPlayStartPosition(-1);
```

Comment:
I know this is not part of the PR, but the negative value of `playStartPositoin` seems to only be used here... To I guess not send a `jumped` signal if stop is pressed multiple times? Idk. If that was the case you could simply check if the value is actually any different in `setTicks` and not emit a signal if the value didn't actually change.

Reply:
No, I understand now it's to not move the playhead back to where it started from if you press stop when it's already stopped. Personally I'd check `isPlaying` instead of using a negative value... but whatever